### PR TITLE
new test set, scratch.ts, dataviz library, and getAverageDistance function + changes to index.ts to calc pieceThatMovedTheFurthest for a set of games

### PR DESCRIPTION
**Pull request by bennyrubanov**

### CODE REVIEW

File: chess-dataviz-1.0.0/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+chess-dataviz
+=============
+
+chess-dataviz is a visualization library for chess, written for D3. It has a number of visualizations to show statistics on games, such as:
```
Comment (EllAchE) [inline chess-dataviz-1.0.0/README.md:4]: @bennyrubanov there are a lot of unrelated changes in this PR, did you intend to add these? Would prefer to separate the review. Looks like you cloned the chess dataviz libary. Any reason we don't just want to use it as a package/import?

**Review by EllAchE — CHANGES_REQUESTED** [review]

**Review by EllAchE — APPROVED** [review]

File: chess-dataviz-1.0.0/README.md (hunk 1 above)
Comment (bennyrubanov) [reply chess-dataviz-1.0.0/README.md:4]: answered in discord

File: src/metrics.ts [hunk 2]
```diff
@@ -136,9 +136,23 @@ export async function getMoveDistanceSingleGame(game: FileReaderGame) {
   return {
     maxDistancePiece,
     maxDistance,
+    distanceMap
   };
 }
 
+export function getAverageDistance(distanceMap: { [key: string]: number }, gameCount: number ) {
```
Comment (EllAchE) [inline src/metrics.ts:143]: can you annotate this function? I'm actually not sure why it exists. Is this for all games or a single game?
